### PR TITLE
[CWS] limit process tree depth when serializing events

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

This PR adds a limit on the depth of the process tree when serializing events.
The current limit is arbitrarily set to 200 processes.

### CODE REVIEW

File: pkg/security/serializers/serializers_linux.go [hunk 1]
```diff
@@ -890,6 +892,13 @@ func newProcessContextSerializer(pc *model.ProcessContext, e *model.Event) *Proc
 		ptr = it.Next()
 	}
 
+	// shrink the middle of the ancestors list if it is too long
+	if len(ps.Ancestors) > processTreeMaxDepth {
+		subLen := processTreeMaxDepth / 2
+		ps.Ancestors = append(ps.Ancestors[0:subLen], ps.Ancestors[len(ps.Ancestors)-subLen-processTreeMaxDepth%2:]...)
```
Comment (spikat) [inline pkg/security/serializers/serializers_linux.go:898]: I'm not sure to understand the last part `-processTreeMaxDepth%2`, `len(ps.Ancestors)-subLen` isn't enough?

Comment (YoannGh) [reply pkg/security/serializers/serializers_linux.go:898]: `-processTreeMaxDepth%2` is only here to handle the case where `processTreeMaxDepth` is an odd number, this is to make sure the depth of the resulting process tree ends up being equal to `processTreeMaxDepth`.
I'll update the PR to add a comment reflecting this :+1:

Comment (spikat) [reply pkg/security/serializers/serializers_linux.go:898]: Got it, thank you ! :bow:

**Review by drichards-87 — APPROVED** [review]

**Review by spikat — APPROVED** [review]
LGTM !

**Pull request merged**